types/src/transaction/mod.rs

```diff
@@ -1138,8 +1218,11 @@ pub struct TransactionOutput {
     /// The amount of gas used during execution.
     gas_used: u64,
 
-    /// The execution status.
+    /// The execution status. The detailed error info will not be stored here instead will be stored in the auxiliary data.
     status: TransactionStatus,
+
+    /// The transaction auxiliary data that includes detail error info that is not used for calculating the hash
+    auxiliary_data: TransactionAuxiliaryData,
```

Comment:
I don't think you can just add fields to `TransactionOutput` -- will instantly break state sync.